scripts/xsslint/xsslint/linters.py

```diff
@@ -439,7 +439,7 @@ def _check_javascript_interpolate(self, file_contents, results):
         """
         Checks that interpolate() calls are safe.
 
-        Only use of StringUtils.interpolate() or HtmlUtils.interpolateText()
+        Only use of StringUtils.interpolate() or HtmlUtils.interpolateHtml()
```

Comment:
`interpolateHtml()` is fine in the test, but I found it confusing here.  Can we just drop it?
```
StringUtils.interpolate() is the only safe `interpolate()` call.
```

Reply:
That entire _check_javascript_interpolate test is confusing. There's no explanation of why we don't want any functions/methods named `interpolate` in javascript.

Reply:
So I did some digging, and it appears that the bare `interpolate()` function there is in fact an i18n helper. [Source here.](https://github.com/open-craft/edx-platform/blob/795c6e44c104b9756b975903f15770d401b637cd/cms/static/js/i18n/am/djangojs.js#L53-L59) It definitely doesn't do any escaping or sanitizing.

But here's where it gets interesting... `StringUtils.interpolate` is defined [here](https://github.com/edx/edx-ui-toolkit/blob/02b0f0d25d021bdf2dca326795ed2cb10546545c/src/js/utils/string-utils.js#L58-L64), and also **does no escaping**. `StringUtils.interpolate`  is just as unsafe as any other `interpolate` function.  The [docs](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss.html#legacy-javascript-files) even mention this:

> ...  then use the plain text StringUtils.interpolate() function as follows. This function will not escape, ...

I vote we remove the entire `_check_javascript_interpolate` lint, because it's confusing and innaccurate.

Reply:
Opened a PR to fix the inconsistencies with the docs: https://github.com/edx/edx-ui-toolkit/pull/109

Reply:
@swalladge: I think you are right about _check_javascript_interpolate.  I think what is missing the JavaScript equivalent of the linting rule [python-interpolate-html](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/preventing_xss/preventing_xss.html#python-interpolate-html).

Reply:
@swalladge: FYI: Looking at the `interpolate` calls in the following file, I think it is clear that the original linting rule was to force people to choose StringUtils vs HtmlUtils going forward, as well as reviewing old code, to make the choice more intentional. See https://github.com/edx/edx-platform/blob/master/lms/templates/commerce/provider.underscore#L21
This is just an FYI.  I still think it is ok for us to have dropped the rule.  In the future, someone just as easily could copy/paste StringUtils and not use HtmlUtils if they don't know what they are doing.